cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/utils/FilterUtils.java

```diff
@@ -111,4 +115,29 @@ public static List<List<String>> cartesianProduct(List<List<String>> orderedValu
             }
         }
     }
+
+    /**
+     * Parses {@link TimeRangeFilter} from spark options.
+     */
+    public static TimeRangeFilter parseSSTableTimeRangeFilter(Map<String, String> options)
+    {
+        if (!options.containsKey(SSTABLE_START_TIMESTAMP_MICROS) && !options.containsKey(SSTABLE_END_TIMESTAMP_MICROS))
+        {
+            return null;
```

Comment:
This is _only_ a nit:

```suggestion
            return TimeRangeFilter.empty(); // create a constant: Range.openClosed(0L, 0L) as empty range in TimeRangeFilter
```